apis/gasbuddy.js

```diff
@@ -0,0 +1,255 @@
+ */
+const { parse } = require('node-html-parser');
+
+const BASE_URL = 'https://www.gasbuddy.com';
+const MAX_PAGE = 1;
+
+let config;
+
+/**
+ * @function getRequestPaths
+ * @description URL paths for fuel type to request data sorted by distance and by price.
+ *
+ * @param {string} type - Fuel type.
+ *
+ * @returns {string[]} URL paths for fuel type.
+ */
+function getRequestPaths(type) {
+    switch(type) {
+        case 'midgrade':
+          type = '2';
+          break;
+        case 'premium':
+          type = '3';
+          break;
+        case 'diesel':
+           type = '4';
+           break;
+        case 'e85':
+           type = '5';
+           break;
+        case 'unl88':
+            type = '12';
+            break;
+        default:
+          type = '1';
+      }
+    return [
+        `/home?search=${config.zip}&fuel=${type}&maxAge=0&method=all`
+    ];
+}
+
+/**
+ * @function mapGasStation
+ * @description Maps HTML gas station to reguilar object.
+ *
+ * @param {Object} htmlGasStation - HTML node of gas station.
+ * @param {string} type - Fuel type.
+ *
+ * @returns {Object} Gas station
+ */
+function mapGasStation(htmlGasStation, type) {
+    return {
+        name: htmlGasStation.querySelector('[class*=header__header3___] a[href*=station]').innerHTML,
+        address: htmlGasStation.querySelector('[class*=StationDisplay-module__address___]').innerHTML.replace('<br>', ' '),
+        prices: { [type]: parseFloat(htmlGasStation.querySelector('[class*=StationDisplayPrice-module__price___]').innerHTML.replace('$', '')) },
+        distance: 0,
+        stationId: htmlGasStation.querySelector('[class*=header__header3___] a[href*=station]').rawAttributes.href.replace('/station/', '')
+    };
+
+}
+
+/**
+ * @function fillMissingPrices
+ * @description Replaces missing price information with max price for type.
+ *
+ * @param {Object} station - Gas Station
+ * @param {Object} maxPricesByType - Maximum price per fuel type.
+ *
+ * @returns {void}
+ */
+function fillMissingPrices(station, maxPricesByType) {
+    for (const type of config.types) {
+        if (!station.prices[type]) {
+            station.prices[type] = `>${maxPricesByType[type]}`;
+        }
+    }
+}
+
+/**
+ * @function sortByPrice
+ * @description Helper function to sort gas stations by price.
+ *
+ * @param {Object} a - Gas Station
+ * @param {Object} b - Gas Station
+ *
+ * @returns {number} Sorting weight.
+ */
+function sortByPrice(a, b) {
+    const aPrice = a.prices[config.sortBy];
+    const bPrice = b.prices[config.sortBy];
+
+    if (!isNaN(aPrice) || !isNaN(bPrice)) {
+        return isNaN(aPrice) ? -1 : 1;
+    }
+
+    return 0;
+}
+
+/**
+ * @function fetchPaginatedStations
+ * @description Paginated API requests for specified type.
+ * @async
+ *
+ * @param {string} type - Fuel type.
+ * @param {string} path - URL path.
+ *
+ * @returns {Promise} Object with type and stations.
+ */
+async function fetchPaginatedStations(type, path) {
```

Comment:
You can simplify this function. You only fetch 1 page (nextPage = 1 and MAX_PAGE = 1) so there is no pagination required.

Reply:
resolved.